service/api_logic/sports_logic.py

```diff
@@ -14,4 +13,32 @@ def get_all_sports():
             "sport_id": sport.sport_id,
             "sport_name": sport.sport_name,
             "sport_img": sport.sport_img
-        } for sport in sports], 200
\ No newline at end of file
+        } for sport in sports], 200
+
+
+@handle_exceptions
+def get_all_leagues_by_sport(session, sport_name, page, per_page):
+    if sport_name is None or "Unknown":
+        return {"error": "Not param sport_name"}
+    offset = (page - 1) * per_page
+    leagues = (
+        session.query(League)
+        .join(Sport, Sport.sport_id == League.sport_id)
+        .filter(Sport.sport_name == sport_name)
+        .limit(per_page)
+        .offset(offset)
+        .all()
+    )
+    if not leagues:
+        print_error_message("No leagues found for the sport")
+        return [], 204
```

Comment:
Same here

Do not return status code.
Since we already went with custom exceptions approach - let's use it. 
Throw a custom exception here as well

Reply:
I do this here because sometimes db can just be empty

Reply:
@Iarynovskyi 

I understand, do the validation for that. But my concern is your "return"

It'd be better to just:

`if not leagues:
   raise SomeCustomExceptionWithCorrectStatusCode(your_message)`